src/remote-desktop.c

```diff
@@ -394,6 +465,7 @@ handle_select_devices (XdpDbusRemoteDesktop *object,
   g_autoptr(GError) error = NULL;
   g_autoptr(XdpDbusImplRequest) impl_request = NULL;
   GVariantBuilder options_builder;
+  GVariant *options;
```

Comment:
Perhaps `g_autoptr(GVariant)` and use `g_steal_pointer()` below?

Reply:
It made things a bit awkward, since we expect it to be consumed by `replace_remote_desktop_restore_token_with_data()` even on error, meaning can't `g_steal_pointer()` it in the argument, but would have to after.

Perhaps it'd be nicer to pass one `in` and one `out` `options` to that function instead.

Reply:
Yeah, I think that's probably better. Not sure why I went with inout, I don't like it :stuck_out_tongue: 

This can be in separate commits btw, or even follow-up PRs, if it complicates things too much.